datalad/distribution/get.py

```diff
@@ -49,53 +51,209 @@
 lgr = logging.getLogger('datalad.distribution.get')
 
 
+def _sort_paths_into_datasets(paths, out=None, dir_lookup=None,
+                              recursive=False, recursion_limit=None):
+    """Returns dict of `existing dataset path`: `directory` mappings
+
+    Any paths that are not part of a dataset or ignored.
+    """
+    # sort paths into the respective datasets
+    if dir_lookup is None:
+        dir_lookup = {}
+    if out is None:
+        out = {}
+    # paths that don't exist (yet)
+    unavailable_paths = []
+    for path in paths:
+        if not lexists(path):
+            # not there yet, impossible to say which ds it will actually
+            # be in, if any
+            unavailable_paths.append(path)
+            continue
+        # the path exists in some shape or form
+        if isdir(path):
+            # this could contain all types of additional content
+            d = path
+        else:
+            # for everything else we are interested in the container
+            d = dirname(path)
+            if not d:
+                d = curdir
+        # this could be `None` if there is no git repo
+        dspath = dir_lookup.get(d, GitRepo.get_toppath(d))
+        dir_lookup[d] = dspath
+        if not dspath:
+            lgr.warning("%s is not part of a dataset, ignored.", path)
+            continue
+        if isdir(path):
+            ds = Dataset(dspath)
+            # we need to doublecheck that this is not a subdataset mount
+            # point, in whic case get_toppath() would point to the parent
+            smpath = ds.get_containing_subdataset(
+                path, recursion_limit=1).path
+            if smpath != dspath:
+                # fix entry
+                dir_lookup[d] = smpath
+                # submodule still needs to be obtained
+                unavailable_paths.append(path)
+                continue
+            if recursive:
+                # make sure we get everything relevant in all _checked out_
+                # subdatasets, obtaining of previously unavailable subdataset
+                # else done elsewhere
+                subs = ds.get_subdatasets(fulfilled=True,
+                                          recursive=recursive,
+                                          recursion_limit=recursion_limit)
+                for sub in subs:
+                    subdspath = opj(dspath, sub)
+                    if subdspath.startswith(_with_sep(path)):
+                        # this subdatasets is underneath the search path
+                        # we want it all
+                        out[subdspath] = [subdspath]
+        out[dspath] = out.get(dspath, []) + [path]
+    return out, unavailable_paths, dir_lookup
+
+
+def _get(content_by_ds, refpath=None, source=None, jobs=None,
+         fulfill_datasets=False):
```

Comment:
so how is `_get` different from the main `get`? i.e. might be nice to get some docstring